src/components/Message/Timestamp.tsx

```diff
@@ -1,48 +1,54 @@
+import clsx from 'clsx';
 import React, { useMemo } from 'react';
 
 import { useMessageContext } from '../../context/MessageContext';
 import { isDate, useTranslationContext } from '../../context/TranslationContext';
-import { getDateString } from '../../i18n/utils';
+import { getDateString, TimestampFormatterOptions } from '../../i18n/utils';
 
-export interface TimestampProps {
-  /* If true, call the `Day.js` calendar function to get the date string to display. */
-  calendar?: boolean;
+export interface TimestampProps extends TimestampFormatterOptions {
   /* Adds a CSS class name to the component's outer `time` container. */
   customClass?: string;
-  /* Overrides the default timestamp format */
-  format?: string;
   /* Timestamp to display */
   timestamp?: Date | string;
 }
 
 export const defaultTimestampFormat = 'h:mmA';
 
 export function Timestamp(props: TimestampProps) {
-  const { timestamp, calendar = false, customClass = '', format = defaultTimestampFormat } = props;
+  const {
+    calendar,
+    calendarFormats,
+    customClass,
+    format = defaultTimestampFormat,
+    timestamp,
+  } = props;
 
   const { formatDate } = useMessageContext('MessageTimestamp');
-  const { tDateTimeParser } = useTranslationContext('MessageTimestamp');
+  const { t, tDateTimeParser } = useTranslationContext('MessageTimestamp');
 
   const normalizedTimestamp = timestamp && isDate(timestamp) ? timestamp.toISOString() : timestamp;
 
   const when = useMemo(
     () =>
       getDateString({
         calendar,
+        calendarFormats,
         format,
         formatDate,
         messageCreatedAt: normalizedTimestamp,
+        t,
         tDateTimeParser,
+        timestampTranslationKey: 'timestamp/Timestamp',
       }),
-    [formatDate, calendar, tDateTimeParser, format, normalizedTimestamp],
+    [calendar, calendarFormats, format, formatDate, normalizedTimestamp, t, tDateTimeParser],
   );
 
   if (!when) {
     return null;
   }
 
   return (
-    <time className={customClass} dateTime={normalizedTimestamp} title={normalizedTimestamp}>
+    <time className={clsx(customClass)} dateTime={normalizedTimestamp} title={normalizedTimestamp}>
```

Comment:
```suggestion
    <time className={customClass} dateTime={normalizedTimestamp} title={normalizedTimestamp}>
```
I don't think this is needed since we're not merging or adding stuff conditionally.